### PR TITLE
Fix virtual reproject geobox crs

### LINKED ISSUE #1072: Need the spatial coords to carry correct `spatial_ref` and `crs`
### Expected behaviour
The spatial coordinates `x` and `y` should have correct `spatial_ref` and `crs` after reproject

### Actual behaviour
Only the values are changed but with the old `spatial_ref` and no `crs`. Hence the `results.geobox` crs is wrong.
https://github.com/opendatacube/datacube-core/blob/6ee0462c5de320d820aa4fdd27a4807fe2ab76c2/datacube/virtual/impl.py#L909-L910

### Steps to reproduce the behaviour
N/A

### Environment information

* Which ``datacube --version`` are you using?
  Open Data Cube core, version 1.8.4.dev52+g07bc51a5
* What datacube deployment/enviornment are you running against?
  Sandbox

### PR DESCRIPTION
**Pull request by emmaai**
### Reason for this pull request
Fix issue #1072 

### Proposed changes

- Populate `coords` with `coords`  from `geobox` with correct `crs` and `spatial_ref`
- Add test from @uchchwhash 

 - [x] Closes #1072 
 - [x] Tests added / passed
 - [x] Fully documented, including `docs/about/whats_new.rst` for all changes

### CODE REVIEW

**Review by uchchwhash — APPROVED** [review]
:+1:

File: datacube/virtual/impl.py [hunk 1]
```diff
@@ -824,8 +827,8 @@ def fetch(self, grouped: VirtualDatasetBox, **load_settings: Dict[str, Any]) ->
         result = xarray.Dataset()
         result.coords['time'] = grouped.box.coords['time']
 
-        for name, coord in grouped.geobox.coordinates.items():
-            result.coords[name] = (name, coord.values, {'units': coord.units, 'resolution': coord.resolution})
+        coords = OrderedDict(**geobox.xr_coords(with_crs=spatial_ref))
+        result.coords.update(coords)
```
Comment (uchchwhash) [inline datacube/virtual/impl.py:831]: should just mention that these two lines can be collapsed into:
```
result.coords.update(geobox.xr_coords(with_crs='spatial_ref'))
```

Comment (uchchwhash) [reply datacube/virtual/impl.py:831]: in which case the collections import won't be needed.

**Review by uchchwhash — APPROVED** [review]

File: datacube/virtual/impl.py (hunk 1 above)
Comment (emmaai) [reply datacube/virtual/impl.py:831]: I've copied those lines from `create_storage` in datacube.api, not sure if it's still doing the right thing if omitting the `OrderedDict`...

**Pull request merged**